types/ibc.go

```diff
@@ -10,9 +10,60 @@ type IBCChannel struct {
 	CounterpartyEndpoint IBCEndpoint `json:"counterparty_endpoint"`
 	Order                IBCOrder    `json:"order"`
 	Version              string      `json:"version"`
-	// optional
-	CounterpartyVersion string `json:"counterparty_version,omitempty"`
-	ConnectionID        string `json:"connection_id"`
+	ConnectionID         string      `json:"connection_id"`
+}
+
+type IBCChannelOpenMsg struct {
+	OpenInit *IBCOpenInit `json:",omitempty"`
+	OpenTry  *IBCOpenTry  `json:",omitempty"`
+}
+
+type IBCOpenInit struct {
```

Comment:
Is there a less verbose way to do this? I tried nested anonymous structs, but those are awful since the whole type signature has to be repeated every time a struct literal is constructed.

Reply:
I don't think there is a better way. If we wanted to have simpler types we had to write custom deserializers and different structures than in Rust. But that would be verbose in a different way.